tests/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriberTest.php

```diff
@@ -0,0 +1,102 @@
+
+        $document = $this->prophesize(ResourceSegmentSubscriberTestDocument::class);
+        $event->getDocument()->willReturn($document->reveal());
+
+        $node = $this->prophesize(NodeInterface::class);
+        $event->getNode()->willReturn($node->reveal());
+
+        $structureMetadata = $this->prophesize(StructureMetadata::class);
+        $inspector->getStructureMetadata($document->reveal())->willReturn($structureMetadata->reveal());
+
+        $property = $this->prophesize(PropertyInterface::class);
+        $property->getName()->willReturn($propertyName);
+        $structureMetadata->getPropertyByTagName('sulu.rlp')->willReturn($property->reveal());
+
+        $inspector->getOriginalLocale($document->reveal())->willReturn($locale);
+        $encoder->localizedSystemName($propertyName, $locale)->willReturn($localizedPropertyName);
+
+        $node->getPropertyValueWithDefault($localizedPropertyName, '')->willReturn($segment);
+
+        // Asserts
+        $document->setResourceSegment($segment)->shouldBeCalled();
+
+        $subscriber = new ResourceSegmentSubscriber($encoder->reveal(), $inspector->reveal());
+        $subscriber->doHydrate($event->reveal());
+    }
+
+    public function testPersist()
+    {
+        $locale = 'de';
+        $segment = '/test';
+        $propertyName = 'url';
+        $localizedPropertyName = sprintf('i18n:%s-%s', $locale, $propertyName);
+
+        $encoder = $this->prophesize(PropertyEncoder::class);
+        $inspector = $this->prophesize(DocumentInspector::class);
+        $event = $this->prophesize(PersistEvent::class);
+
+        $document = $this->prophesize(ResourceSegmentSubscriberTestDocument::class);
+        $event->getDocument()->willReturn($document->reveal());
+
+        $structureMetadata = $this->prophesize(StructureMetadata::class);
+        $inspector->getStructureMetadata($document->reveal())->willReturn($structureMetadata->reveal());
+
+        $property = $this->prophesize(PropertyInterface::class);
+        $property->getName()->willReturn($propertyName);
+        $structureMetadata->getPropertyByTagName('sulu.rlp')->willReturn($property->reveal());
+
+        $localizedProperty = $this->prophesize(PropertyInterface::class);
+        $localizedProperty->getName()->willReturn($localizedPropertyName);
+
+        $structure = $this->prophesize(StructureInterface::class);
+        $structure->getProperty($propertyName)->willReturn($localizedProperty->reveal());
+        $document->getStructure()->willReturn($structure->reveal());
+
+        $document->getResourceSegment()->willReturn($segment);
+
+        // Asserts
+        $localizedProperty->setValue($segment)->shouldBeCalled();
+
+        $subscriber = new ResourceSegmentSubscriber($encoder->reveal(), $inspector->reveal());
+        $subscriber->doPersist($event->reveal());
+    }
+}
+
+interface ResourceSegmentSubscriberTestDocument extends ResourceSegmentBehavior, StructureBehavior
```

Comment:
I think this is not necessary, since you can also code something like this:

```php
$document = $this->prophesize(ResourceSegmentBehavior::class);
$document->willImplement(StructureBehavior::class);
```